src/main/java/com/stripe/model/BalanceTransaction.java

```diff
@@ -62,10 +62,20 @@ public void setSourceObject(HasId o) {
     this.source = new ExpandableField<HasId>(o.getId(), o);
   }
 
+  /**
+   * Returns the {@code source} object, cast to the provided type parameter.
+   *
+   * @deprecated Use {@link #getSourceObjectAs(Class)} instead.
+   */
+  @Deprecated
   @SuppressWarnings("unchecked")
   public <O extends HasId> O getSourceObjectAs() {
     return (this.source != null) ? (O) this.source.getExpanded() : null;
   }
+
+  public <O extends HasId> O getSourceObjectAs(Class<O> clazz) {
```

Comment:
Doesn't this reduce the usefulness of this method? As in you used to be able to invoke it with just a type parameter, but now you need another parameter as well?

Reply:
I guess -- tbh I have a hard time wrapping my head around Java's generics so I'm not sure if this new method actually offers additional safety, or it just pleases ErrorProne for no benefit...

I'll remove this commit from the PR so that we can merge the rest of the fixes, we can always revisit this particular issue later.

Reply:
Sounds good. I was just [reading through the documentation on this lint](https://github.com/google/error-prone/blob/master/docs/bugpattern/TypeParameterUnusedInFormals.md), but it doesn't seem to quite adequately explain all ramifications — they talk about why it's not type safe, which is useful, but they don't explain the obvious hit on usability. Looking at a method like this one, I would always have assumed that there is some runtime risk of the cast failing.